### PR TITLE
cudagraph_trees: honor mark_static_address outputs

### PR DESCRIPTION
**Pull request by tfogal**
cudagraph_trees classifies each storage that is live at the end of capture as either a private-pool allocation or a known static input. An output whose storage was flagged via torch._dynamo.mark_static_address was neither, so check_memory_pool reported it as "not allocated in pool but should be", and (with the slow-path asserts off) the pool manager poisoned/freed it across steps ("accessing tensor output of CUDAGraphs that has been overwritten by a subsequent run").

mark_static_address marks a tensor's address as constant/externally- managed; this honors that on outputs too. An output carrying _dynamo_static_input_type (renamed to just _dynamo_static_type) is now treated as a static output: excluded from the warmup live-storage set (add_ref) and classified as a static output at record time (_add_first_outputs), so it skips the pool-membership check and per-step deallocation -- exactly as an output aliasing a persistent static input already does.

Authored with Claude.

Note to reviewers: the core of this is the changes to `torch/_inductor/cudagraph_trees.py`. The other changes are cosmetic: it made sense to rename `_dynamo_static_input_type` to just `_dynamo_static_type` given that it's no longer limited to just inputs, and then other renames cascaded from there. Happy to remove the cosmetic aspects.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude explain those CI failures, check if they are related to this PR, and review the changes

File: test/inductor/test_cudagraph_trees.py [hunk 1]
```diff
@@ -1886,6 +1886,75 @@ def foo(args):
...
+        def test_static_address_output_classified_as_static(self):
+            # Outputs marked as a static address should have that state
+            # propagate through the compiler.
+            pool = torch.cuda.MemPool()
+            with torch.cuda.use_mem_pool(pool):
+                static_out = torch.zeros(20, 20, device="cuda")
+            torch._dynamo.mark_static_address(static_out)
+
+            def foo(args):
+                x = args[0]
```
Comment (eellison) [inline test/inductor/test_cudagraph_trees.py:1898]:
```
  def test_static_address_output_compile_existing_path(self):
      pool = torch.cuda.MemPool()
      with torch.cuda.use_mem_pool(pool):
          static_out = torch.zeros(20, 20, device="cuda")
      torch._dynamo.mark_static_address(static_out)

      @torch.compile(mode="reduce-overhead", fullgraph=True)
      def f(x):
          static_out.copy_(x + 1)
          return static_out

      for gen in range(4):
          inp = torch.full((20, 20), float(gen), device="cuda")
          self.assertEqual(f(inp), inp + 1)

      node = self.curr_node()
      self.assertEqual(list(node.wrapped_function.static_input_idxs), [1])
      self.assertIs(node.static_output_tensors[0], static_out)
      self.assertIsNone(node.outputs_weakrefs[0])
```

Passes. do you have a repro that doesnt ? im not sure i follow what the issue is.
